### PR TITLE
Fixes for make errors.

### PR DESCRIPTION
**Pull request by ivmarkp**
@samuelharden Please verify the changes that I made to fix make errors. Might give us a clue for why indexing is still failing with segmentation fault error on my system. My best guess is that something is odd in `trec_index.cc` that might be causing indexing to fail. It successfully decompresses documents in directory specified by `textfile` parameter i.e. `./en.docs.2011/en_BDNews24/1/`

### CODE REVIEW

File: src/trec_index.cc [hunk 1]
```diff
@@ -352,7 +352,7 @@ int main(int argc, char **argv)
     // Catch any Xapian::Error exceptions thrown
     try {
         // Make the database
-        Xapian::WritableDatabase db(Xapian::Brass::open(trec_config.get_db().c_str(), Xapian::DB_CREATE_OR_OPEN));
+        Xapian::WritableDatabase db("/home/vivek/Downloads/FIREData/2011/en.docs.2011/index_business/", Xapian::DB_CREATE_OR_OPEN);
```
Comment (samuelharden) [inline src/trec_index.cc]: @ivmarkp Not a good idea to hardcode the path here

File: src/trec_search.cc [hunk 2]
```diff
@@ -61,8 +61,7 @@ int load_query( std::ifstream & queryfile, int & topicno,Xapian::Query & query,
 	Xapian::SimpleStopper stopper;
 	get_stopper(stopper);
 	qp.set_stopper(&stopper);
-	qp.set_stemmer(stemmer);
```
Comment (samuelharden) [inline src/trec_search.cc]: Stemmer should still be valid. We should set stemmer

File: src/trec_search.cc [hunk 3]
```diff
@@ -156,15 +156,6 @@ int main(int argc, char **argv)
 			enquire.set_weighting_scheme(Xapian::TradWeight());
 		}
  	}
-	else if (config.use_weightingscheme("lmweight")) {
```
Comment (samuelharden) [inline src/trec_search.cc]: We were not suppose to remove LMWeight Weighting scheme. Scheme has changed specifically one of the field was removed. We had to adjust the constructor

File: src/trec_search.cc [hunk 4]
```diff
@@ -61,8 +61,7 @@ int load_query( std::ifstream & queryfile, int & topicno,Xapian::Query & query,
 	Xapian::SimpleStopper stopper;
 	get_stopper(stopper);
 	qp.set_stopper(&stopper);
-	qp.set_stemmer(stemmer);
-	qp.set_bigram(config.get_queryparsebigram());  
+	qp.set_stemmer(stemmer);
```
Comment (ivmarkp) [inline src/trec_search.cc]: Stemmer is set here.

**Comment by ivmarkp** [comment]
All issues were fixed eventually after discussions on IRC so I think it should be fine to close this PR. Thanks!

**Pull request closed without merging**